subnet/deployment-generator/docker/Dockerfile

```diff
@@ -1,20 +1,27 @@
 
 FROM golang:1.14 as builder
 
-#use node base image since it can already run the go binary
-# RUN curl -fsSL https://deb.nodesource.com/setup_20.x | bash - &&\  
-#     apt-get install -y nodejs
-# RUN apt-get update && apt-get install -y build-essential 
-# RUN npm install -g yarn
+ARG SUBNET_BRANCH=master    
 
 RUN git clone https://github.com/XinFinOrg/XDC-Subnet
 
-ARG SUBNET_BRANCH=master        
 #it is called 'BRANCH' but commits also work
 RUN cd XDC-Subnet && git checkout ${SUBNET_BRANCH} &&  make puppeth
 
+
+
+
+
 FROM node:18.15
 
+ARG SUBNET_BRANCH=master    
+ARG IMAGE_NAME=latest
+#inject build arg as ENV
+ENV SUBNET_BRANCH=${SUBNET_BRANCH}
+ENV IMAGE_NAME=${IMAGE_NAME}
+RUN echo SUBNET_BRANCH=${SUBNET_BRANCH}
+RUN echo IMAGE_NAME=${IMAGE_NAME}
+
```

Comment:
why we need these?

Reply:
generator clones the subnet repo for puppeth, so this is the way to record the subnet's branch/commit within this image. 

the IMAGE_NAME is for the generated command to self reference its version, ex. the generator-v3 tag instead of latest. 

`  docker run --env-file generated/common.env \
    -v $(pwd)/generated/:/app/generated/ \
    --entrypoint 'bash' xinfinorg/subnet-generator:generator-v3 ./deploy_csc.sh `

Reply:
the args are injected at build-time, by github actions